### DISCUSSION TITLE
Customize KtdGridLayout

### DISCUSSION THREAD

**Discussion by Ale11Nishitha** [katoid/angular-grid-layout#53] [Q&A]
I have a requirement in which I have to customize KtdGridLayout interface and on Layout update I want those customized changes. Upon customization  I want those changes in npm i . Is there any way ?

**Comment by llorenspujol** [comment]
If those changes make sense they can be included in the project. What are those changes about? Can you make a PR to see them?
Thanks!

**Reply by Ale11Nishitha** [reply]
Thanks for the reply !

export interface KtdGridLayoutItem {
    id: string;
    x: number;
    y: number;
    w: number;
    h: number;
    minW?: number;
    minH?: number;
    maxW?: number;
    maxH?: number;
    routerLink?:"/shared/todo",
    icon?:"../../../../assets/images/todolist.png",
    widgetName?:"To Do List"
    
}

This is how I want my interface. Is there any way to pass custom data ?

**Comment by llorenspujol** [comment]
`KtdGridLayoutItem` should have only the specified grid data. Is true that we could add some property data like `data: any` or` data: T` to this interface, but I prefer another way to work (for now).

The recommended way to handle this scenario right now, is to have a Dictionary or a List with the extra data you need. You afterwards can retrieve the item data you need by id.

I hope this solve your issue, and if not, tell me exactly why this solution does not work and we will find a way.
